python/lsst/ip/diffim/zogy.py

```diff
@@ -785,26 +785,31 @@ def makeDiffimSubExposure(self, ftDiff):
 
         if ftDiff.S is not None:
             S = self.inverseFftAndCropImage(
-                ftDiff.S, self.imgShape, np.logical_or(self.fftFullIm1.filtInf, self.fftFullIm2.filtInf),
-                np.logical_or(self.fftFullIm1.filtNaN, self.fftFullIm2.filtNaN),
-                dtype=self.subExposure1.image.dtype)
+                ftDiff.S, self.imgShape, dtype=self.subExposure1.image.dtype)
             varPlaneS = self.inverseFftAndCropImage(
-                ftDiff.varPlaneS, self.imgShape,
-                np.logical_or(self.fftVarPl1.filtInf, self.fftVarPl2.filtInf),
-                np.logical_or(self.fftVarPl1.filtNaN, self.fftVarPl2.filtNaN),
-                dtype=self.subExposure1.variance.dtype)
+                ftDiff.varPlaneS, self.imgShape, dtype=self.subExposure1.variance.dtype)
+
+            # There is no detection where the image or its variance was not finite
+            flt = np.logical_or(self.fftFullIm1.filtInf, self.fftFullIm2.filtInf)
+            flt = np.logical_or(flt, self.fftFullIm1.filtNaN)
+            flt = np.logical_or(flt, self.fftFullIm2.filtNaN)
+            flt = np.logical_or(flt, self.fftVarPl1.filtInf)
+            flt = np.logical_or(flt, self.fftVarPl2.filtInf)
+            flt = np.logical_or(flt, self.fftVarPl1.filtNaN)
+            flt = np.logical_or(flt, self.fftVarPl2.filtNaN)
+            tiny = np.finfo(varPlaneS.dtype).tiny * 100
+            # Ensure that no division by 0 occurs in S/sigma(S).
+            # S is set to be always finite, 0 where pixels non-finite
+            flt = np.logical_or(flt, varPlaneS < tiny)
+            varPlaneS[flt] = 1
```

Comment:
Indeed, 0. / tiny == 0. , but this is true so long we keep the same dtype precision. Exposures can be single floats; as float32(double_tiny) == 0. this would yield 0. / 0. in  S/sigma(S) when copied into an `Exposure`. Now, we could have used the "typical" variance level here but it's not cheaply available. If we go with a constant, I think 1. is the most meaningful here.